.github/workflows/end-to-end.yaml

```diff
@@ -0,0 +1,103 @@
+name: End-to-End Tests
+on:
+  # Run once a day, every day.
+  schedule:
+    - cron:  '0 0 * * *'
+
+jobs:
+  download-linux:
+    strategy:
+      matrix:
+        tree: ["torvalds_linux", "bpf_bpf-next"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Retrieve Linux sources
+        run: |
+          name=${{ matrix.tree }}
+          suffix=${name//_/\/}
+          tree=git://git.kernel.org/pub/scm/linux/kernel/git/$suffix.git
+          git clone -qb master --depth 1 $tree linux
+
+      - name: Compress Linux sources
+        run: |
+          tar -zc linux -f linux.tgz
+
+      - name: Upload Linux sources
```

Comment:
As I understand, this zips the linux sources and uploads them as an artifacts that can be downloaded (called `bpf_bpf-next` and `torvalds_linux`). Is the purpose to be able to quickly download and check locally, as opposed to cloning locally? 

Only concern is the sources when compressed are ~500GB. Not sure if github has a limit on storage for artifacts, but I understand there's a time limit (90 days).

Reply:
Oh, that's a good question. That should definitely be documented in the commit description. I will update it before merging.

In short, I needed to do this to avoid having 66 jobs trying to download Linux sources in parallel. If they try, they get throttled and all fail. So instead I download from only two initial jobs then share with the subsequent jobs. I explain that in the commit.

I'm unsure if there's any limit on size of artifacts. Worst case, we can optimize that by deleting the artifacts after the second set of jobs downloaded them. There isn't a straightforward way to do it in GitHub workflows so I skipped it for the time being, but it seems possible.